src/core/src/FabricUtil.cpp

```diff
@@ -497,4 +497,38 @@ std::string printFabricStage() {
     return stream.str();
 }
 
+FabricStatistics getStatistics() {
+    FabricStatistics statistics;
+
+    auto sip = UsdUtil::getFabricStageInProgress();
+
+    const auto geometryBuckets = sip.findPrims(
+        {carb::flatcache::AttrNameAndType(FabricTypes::_cesium_tilesetId, FabricTokens::_cesium_tilesetId)},
+        {carb::flatcache::AttrNameAndType(FabricTypes::Mesh, FabricTokens::Mesh)});
+
+    const auto materialBuckets = sip.findPrims(
+        {carb::flatcache::AttrNameAndType(FabricTypes::_cesium_tilesetId, FabricTokens::_cesium_tilesetId)},
+        {carb::flatcache::AttrNameAndType(FabricTypes::Material, FabricTokens::Material)});
+
+    for (size_t bucketId = 0; bucketId < geometryBuckets.bucketCount(); bucketId++) {
+        auto paths = sip.getPathArray(geometryBuckets, bucketId);
+        statistics.numberOfGeometriesLoaded += paths.size();
+
+        auto worldVisibilityFabric =
+            sip.getAttributeArrayRd<bool>(geometryBuckets, bucketId, FabricTokens::_worldVisibility);
+        for (size_t i = 0; i < worldVisibilityFabric.size(); i++) {
+            if (worldVisibilityFabric[i]) {
+                statistics.numberOfGeometriesVisible++;
+            }
+        }
```

Comment:
NIT: This seems like a good candidate for [std::accumulate](https://en.cppreference.com/w/cpp/algorithm/accumulate).

Reply:
`std::count` also does the trick